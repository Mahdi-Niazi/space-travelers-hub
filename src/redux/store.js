import { configureStore } from '@reduxjs/toolkit';

import missionReducer from './slices/MissionsSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
