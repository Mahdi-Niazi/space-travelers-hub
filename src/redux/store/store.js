import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from '../slices/RocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

export default store;
