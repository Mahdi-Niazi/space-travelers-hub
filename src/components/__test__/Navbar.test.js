import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../NavLink';
import store from '../../redux/store';

describe('Test the Mission file', () => {
  it('should match to snapshoot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should return the number of navlinks', () => {
    render(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    );
    const navlinks = screen.getAllByRole('link');
    expect(navlinks).toHaveLength(4);
  });
});
