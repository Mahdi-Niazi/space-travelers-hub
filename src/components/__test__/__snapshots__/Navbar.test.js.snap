// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Mission file should match to snapshoot 1`] = `
<nav
  className="Navbar"
>
  <div
    className="nav-container"
  >
    <a
      aria-current="page"
      className="nav-logo active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo"
        src="logo.png"
      />
      <h1>
        Space Travellers
      </h1>
    </a>
    <ul
      className="nav-items"
    >
      <li>
        <a
          aria-current="page"
          className="navlinks active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          Rockets
        </a>
      </li>
      <li
        className="li-missions"
      >
        <a
          className="navlinks-missions"
          href="/missions"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="navlinks"
          href="/my-profile"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
